File: python/src/aegis/resources/reports.py
"""Reports resource — V3 report generation endpoint.

Server contracts:

- ``POST /v3/reports/generate`` — generate a report for the configured period.
"""

from __future__ import annotations

from typing import Any, Dict

from aegis.resources._base import AsyncResource, SyncResource


class SyncReports(SyncResource):

    def generate(self, body: Dict[str, Any]) -> Dict[str, Any]:
        return self._post("/v3/reports/generate", json=body)


class AsyncReports(AsyncResource):

    async def generate(self, body: Dict[str, Any]) -> Dict[str, Any]:
        return await self._post("/v3/reports/generate", json=body)

File: go/reports.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// ReportsService — V3 report generation.
type ReportsService struct {
	t *internal.Transport
}

func (s *ReportsService) Generate(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/reports/generate", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}
